saleor/checkout/forms.py

```diff
@@ -1,18 +1,42 @@
 from django import forms
-from django.contrib.auth.models import AnonymousUser
 from django.utils.translation import ugettext_lazy as _
+from saleor.userprofile.models import Address
 
-from ..userprofile.forms import AddressForm
 
+class AddressChoiceField(forms.ModelChoiceField):
+    choice_values = []
 
-class ShippingForm(AddressForm):
+    def validate(self, value):
+        if value not in self.choice_values:
+            return super(AddressChoiceField, self).validate(value)
 
-    shipping_same_as_billing = forms.BooleanField(initial=True, required=False)
+    def to_python(self, value):
+        if value in self.choice_values:
+            return value
+        else:
+            return super(AddressChoiceField, self).to_python(value)
 
+    def add_magic_choices(self, magic_choices, queryset):
+        choices = list(queryset) if queryset else []
+        self.choices = magic_choices + choices
 
-class DeliveryForm(forms.Form):
 
-    method = forms.ChoiceField(label=_('Shipping method'))
+class UserAddressesForm(forms.Form):
+    address = AddressChoiceField(
+        queryset=Address.objects.none(),
+        widget=forms.RadioSelect)
+
+    def __init__(self, queryset, possibilities, *args, **kwargs):
+        super(UserAddressesForm, self).__init__(*args, **kwargs)
+        address_field = self.fields['address']
+        address_field.queryset = queryset if queryset else Address.objects.none()
+        address_field.add_magic_choices(possibilities, queryset)
+        address_field.choice_values = [value for value, label in possibilities]
```

Comment:
Have the field initialize itself. The class should encapsulate its logic, not depend on an external entity settings its internals.